buidl/test/test_musig.py

```diff
@@ -69,52 +228,86 @@ def test_single_leaf_multisig_locktime(self):
-            tx_obj.get_sig_taproot(0, priv, ext_flag=1) for priv in private_keys[:3]
-        ]
-        tx_obj.finalize_p2tr_multisig(0, sigs)
+        for input_index, pubkeys in enumerate(combinations(points, 3)):
+            tx_in = tx_ins[input_index]
+            tx_in.witness.items = []
+            tx_obj.initialize_p2tr_multisig(
+                input_index,
+                tap_root_input.control_block(leaf),
+                tap_root_input.tap_node.tap_script,
+            )
+            sigs = []
+            for priv in private_keys:
+                if priv.point in pubkeys:
+                    sig = tx_obj.get_sig_taproot(input_index, priv, ext_flag=1)
+                else:
+                    sig = b""
+                sigs.append(sig)
+            tx_obj.finalize_p2tr_multisig(input_index, sigs)
+            self.assertTrue(tx_obj.verify_input(input_index))
+        input_index = len(tx_ins) - 1
+        tx_in = tx_ins[input_index]
+        musig = MuSigTapScript(points)
+        nonce_secret_pairs = []
+        nonce_point_pairs = []
+        for _ in points:
+            nonce_secrets, nonce_points = musig.generate_nonces()
+            nonce_secret_pairs.append(nonce_secrets)
+            nonce_point_pairs.append(nonce_points)
+        sig_hash = tx_obj.sig_hash(input_index, SIGHASH_DEFAULT)
+        nonce_sums = musig.nonce_sums(nonce_point_pairs)
+        r = musig.compute_r(nonce_sums, sig_hash)
+        s_sum = 0
+        for nonce_secrets, priv in zip(nonce_secret_pairs, private_keys):
+            k = musig.compute_k(nonce_secrets, nonce_sums, sig_hash)
+            s_sum += musig.sign(priv, k, r, sig_hash, tweak=tap_root_input.tweak)
+        schnorr = musig.get_signature(s_sum, r, sig_hash, tap_root_input.tweak)
+        tx_in.finalize_p2tr_keypath(schnorr.serialize())
+        self.assertTrue(tx_obj.verify_input(input_index))
+        self.assertEqual(tx_obj.vbytes(), tx_obj.fee())
         self.assertTrue(tx_obj.verify())
-
         tx_obj.locktime = Locktime(current_locktime - 1)
-        tx_in.witness.items = []
-        tx_obj.initialize_p2tr_multisig(
-            0,
-            tap_root_input.control_block(leaf),
-            tap_root_input.tap_node.tap_script,
-        )
-        sigs = [
-            tx_obj.get_sig_taproot(0, priv, ext_flag=1) for priv in private_keys[:3]
-        ]
-        tx_obj.finalize_p2tr_multisig(0, sigs)
-        self.assertFalse(tx_obj.verify())
+        for input_index, pubkeys in enumerate(combinations(points, 3)):
+            tx_in = tx_ins[input_index]
+            tx_in.witness.items = []
+            tx_obj.initialize_p2tr_multisig(
+                input_index,
+                tap_root_input.control_block(leaf),
+                tap_root_input.tap_node.tap_script,
+            )
+            sigs = []
+            for priv in private_keys:
+                if priv.point in pubkeys:
+                    sig = tx_obj.get_sig_taproot(input_index, priv, ext_flag=1)
```

Comment:
Named arguments always better for complex methods, especially when it comes to signing logic.